src/main/clojure/clara/rules/engine.cljc

```diff
@@ -1700,10 +1704,10 @@
 (defn variables-as-keywords
   "Returns symbols in the given s-expression that start with '?' as keywords"
   [expression]
-  (into #{} (for [item (tree-seq coll? seq expression)
-                  :when (and (symbol? item)
-                             (= \? (first (name item))))]
-              (keyword item))))
+  (into #{} (platform/eager-for [item (tree-seq coll? seq expression)
```

Comment:
@ris5438 I'm OK with leaving this for the sake of the pattern of using eager-for in the engine namespace, but just FYI putting the contents of a LazySeq in a set as here will force realization of the lazy sequence anyway.  Also variables-as-keywords takes s-expression input, so it isn't exposed to the mutability problems in the memory that prompted the overall shift to eager-for.